### PR TITLE
Enable profiling support

### PR DESCRIPTION
**Pull request by weapdiv-david**
**_What**_

Enables profiling of the application

**_How to test**_

1) Check out the branch
2) Set the `EQ_PROFILING` environment varable to `True`
3) Run the application as usual
4) Check for the presence of a `profiling` directory containing one file per request.

**_Who can test**_

Anybody except @weapdiv-david

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,7 +181,14 @@ There is a python script for generating tokens for use in development, to run:
 python token_generator.py
 ```
 
+## profiling
 
+Setting the `EQ_PROFILING` environment variable to `True` will enable profiling of the application.  Profiling information
```
Comment (warrenbailey) [inline README.md:187]: We probably should add env this to the environment variable section too.

File: app/__init__.py [hunk 2]
```diff
@@ -139,4 +140,21 @@ def filter(self, record):
...
     application.logger
 
+    # Setup profiling
+    if settings.EQ_PROFILING:
+        from werkzeug.contrib.profiler import ProfilerMiddleware, MergeStream
+        import os
+
+        profiling_dir = "profiling"
+
+        f = open('profiler.log', 'w')
```
Comment (warrenbailey) [inline app/__init__.py:165]: These file names and directories could be a constant.

File: app/__init__.py [hunk 3]
```diff
@@ -139,4 +140,21 @@ def filter(self, record):
     application.logger_name = "nowhere"
     application.logger
 
+    # Setup profiling
```
Comment (warrenbailey) [inline app/__init__.py:158]: This method is getting pretty long now, I wonder if it's worth splitting it out into sub-functions, i.e. one for profiling, one for logging etc. Might make it easier to read?

**Comment by warrenbailey** [comment]
Only really minor comments, looks good though :+1:

File: README.md (hunk 1 above)
Comment (weapdiv-david) [inline README.md:187]: Have updated that

**Comment by warrenbailey** [comment]
LGTM - will refactor app/**init**.py as part of a wider tidy up

**Pull request merged**